### PR TITLE
Propogate client query arguments to join tables

### PR DESCRIPTION
**Pull request by rishi-aga**
One of the PR that Resolves https://github.com/yahoo/elide/discussions/1829

## Description
Propagate client query arguments to join tables
Refactor: Getting column projection from queryable's source itself, this column projection will have default/available arguments for the column.  So no need to go-to source for default/available arguments again.

## License
I confirm that this contribution is made under an Apache 2.0 license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/metadata/ColumnContext.java [hunk 1]
```diff
@@ -97,14 +98,12 @@ public Object get(Object key) {
...
+        ColumnProjection column = this.getQueryable().getSource().getColumnProjection(keyStr);
         if (column != null) {
 
             ColumnProjection newColumn = column.withArguments(
-                            getColumnArgMap(this.getQueryable(),
-                                            column.getName(),
-                                            this.getColumn().getArguments(),
-                                            emptyMap()));
+                            mergedArgumentMap(column.getArguments(),
+                                              this.getColumn().getArguments()));
```
Comment (rishi-aga) [inline elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/metadata/ColumnContext.java:106]: Getting column projection from queryable's source itself, this column projection will have default/available arguments for the column.  So no need to go-to source for default/available arguments again.

File: elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/metadata/ColumnContext.java [hunk 2]
```diff
@@ -233,4 +229,9 @@ private Object resolveSQLHandlebar(final Object context, final Options options)
 
         return columnArgMap;
     }
+
+    public static Map<String, Argument> mergedArgumentMap(Map<String, Argument> referencedColumnArgs,
```
Comment (aklish) [inline elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/metadata/ColumnContext.java:233]: javadoc

Comment (aklish) [reply elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/metadata/ColumnContext.java:233]: These function parameters use terms like column but the function is also leveraged for table arguments.  Let's rename these parameters so they make sense in both contexts.

File: elide-datastore/elide-datastore-aggregation/src/test/java/com/yahoo/elide/datastores/aggregation/integration/MetaDataStoreIntegrationTest.java [hunk 3]
```diff
@@ -200,10 +200,10 @@ public void tableTest() {
         // Verify Table Arguments
         given()
                 .accept("application/vnd.api+json")
-                .get("/table/SalesNamespace_orderDetails?include=arguments")
+                .get("/table/SalesNamespace_orderDetails?include=argumentDefinitions")
```
Comment (aklish) [inline elide-datastore/elide-datastore-aggregation/src/test/java/com/yahoo/elide/datastores/aggregation/integration/MetaDataStoreIntegrationTest.java:203]: The contract of the API shouldn't change here (should still be argument) even though we changed the model name.  The `@Include` type is still 'argument'.  Not sure why this test is passing.

**Review by aklish — CHANGES_REQUESTED** [review]

**Review by aklish — APPROVED** [review]

**Pull request merged**